### ISSUE TITLE
Unable to connect error, Meshcentral Router...

### ISSUE THREAD

**Issue by granad2002** [Ylianst/MeshCentral#5433]
**Describe your issue**
Hi guys, I am getting the following error when I try to connect to a Windows machine via RDP and Meshcentral Router:

![Unbenannt](https://github.com/Ylianst/MeshCentral/assets/81970211/945bc231-682c-4d13-82e0-cbf878e54905)

In syslog I am getting the following error:

```RangeError: Invalid WebSocket frame: MASK must be set
at Receiver.getInfo (.../node_modules/express-ws/node_modules/ws/lib/receiver.js:284:16)
at Receiver.startLoop (.../node_modules/express-ws/node_modules/ws/lib/receiver.js:131:22)
at Receiver._write (.../node_modules/express-ws/node_modules/ws/lib/receiver.js:78:10)
at writeOrBuffer (node:internal/streams/writable:389:12)
at _write (node:internal/streams/writable:330:10)
at Receiver.Writable.write (node:internal/streams/writable:334:10)
at Socket.socketOnData (.../node_modules/express-ws/node_modules/ws/lib/websocket.js:1162:35)
at Socket.emit (node:events:390:28)
at addChunk (node:internal/streams/readable:315:12)
at readableAddChunk (node:internal/streams/readable:289:9)
at Socket.Readable.push (node:internal/streams/readable:228:10)
at TCP.onStreamRead (node:internal/stream_base_commons:199:23) {
code: 'WS_ERR_EXPECTED_MASK',
[Symbol(status-code)]: 1002
```

It was working for years therefore I think it can be related to the newest version 1.1.13.

Does someone have an idea what the problem is?

Thanks in advance

**Server Software (please complete the following information):**
 - OS: Ubuntu
 - Virtualization: 
 - Network: Reverse Proxy nginx

**Client Device (please complete the following information):**
 - Device:
 - OS: Windows
 - Network: Remote over WAN
 - Browser: Firefox
 - MeshCentralRouter Version: see screenshot

**Comment by si458** [comment]
you have missed your config.json file for us to help you.
are you using cloudflare? - https://github.com/Ylianst/MeshCentral/issues/5302

**Comment by granad2002** [comment]
Yes, thx, would be great if you can have a look at it.

Sorry for missing the config.json, but it should have nothing to do with it as I didn't change anything there and it was working before installing the newest version.

And no, I am not using Cloudflare.

**Comment by si458** [comment]
@granad2002 its all working fine here so you are gunna have to share ur config.json plz so i can try replicate ur setup
what node js version are you using?

**Comment by granad2002** [comment]
I am using node js version 16.13.1. I've forgotten to add that I am using nginx as reverse proxy.

Here is my config.json:

```{
  "$schema": "http://info.meshcentral.com/downloads/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer >
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "cert": "XXX",
    "TrustedCert": false,
    "_WANonly": true,
    "_LANonly": true,
    "_sessionKey": "XXX",
    "port": 4430,
    "relayDNS":[ "XXX", "XXX", "XXX", "XXX", "XXX"],
    "_relayAliasPort": 443,
    "aliasPort": 443,
    "redirPort": 800,
    "AgentPong": 300,
    "TlsOffload": "127.0.0.1",
    "plugins": { "enabled": true },
    "_redirAliasPort": 80,
    "AllowLoginToken": true,
    "AllowFraming": true,
    "mstsc": true
  },
  "domains": {
    "": {
      "certUrl": "https://127.0.0.1:443/",
      "userQuota": 5000000000,
      "meshQuota": 5000000000,
      "title": "MeshCentral",
      "title2": "XXX",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true
    }
  },
  "letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "XXX",
    "names": "XXX",
    "rsaKeySize": 3072,
    "production": true
  }
}
`

**Comment by si458** [comment]
have u tried downloading the new version from the web ui or from github directly and using that?
https://github.com/Ylianst/MeshCentral/raw/master/agents/MeshCentralRouter.exe

**Comment by mcallisaya79** [comment]
Using the latest version, I can login now, but the RDP option is not working, it does nothing (tested on different computers), map relays are working fine, but RDP is doing nothing.

ExceptionSink: System.NullReferenceException: Object reference not set to an instance of an object.
   at MeshCentralRouter.MainForm.QuickMap(Int32 protocol, Int32 port, Int32 appId, NodeClass node)
   at MeshCentralRouter.MainForm.rdpToolStripMenuItem_Click(Object sender, EventArgs e)
   at System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
   at System.Windows.Forms.ToolStripMenuItem.OnClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
   at System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
   at System.Windows.Forms.ToolStripDropDown.OnMouseUp(MouseEventArgs mea)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ToolStrip.WndProc(Message& m)
   at System.Windows.Forms.ToolStripDropDown.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


![image](https://github.com/Ylianst/MeshCentral/assets/144259718/8ee2d73f-2c5f-4317-9c6b-bcfad5f60da4)

**Comment by granad2002** [comment]
> have u tried downloading the new version from the web ui or from github directly and using that? https://github.com/Ylianst/MeshCentral/raw/master/agents/MeshCentralRouter.exe

Thx, I tried that now and what should I say, it works! I already tried that sometime this week but Windows wouldn't let me start the .exe because of the Smartscreen filter or something. But now it worked.

**Comment by si458** [comment]
> > have u tried downloading the new version from the web ui or from github directly and using that? https://github.com/Ylianst/MeshCentral/raw/master/agents/MeshCentralRouter.exe
> 
> Thx, I tried that now and what should I say, it works! I already tried that sometime this week but Windows wouldn't let me start the .exe because of the Smartscreen filter or something. But now it worked.

@Ylianst updated the meshcentralrouter literally last week, so it might be it wasn't included in the 1.1.13 update or it's not detecting the new version correctly!

Also yeh the smartscreen is because I think again @Ylianst certificate has expired (these aren't cheap anymore!)
